monai/data/decathlon_datalist.py

```diff
@@ -148,3 +149,40 @@ def load_decathlon_properties(data_property_file_path: str, property_keys: Union
             raise KeyError(f"key {key} is not in the data property file.")
         properties[key] = json_data[key]
     return properties
+
+
+def check_missing_files(
+    datalist: List[Dict], keys: KeysCollection, root_dir: Optional[str] = None, allow_missing_keys: bool = False
+):
+    """Checks whether some files in the Decathlon datalist are missing.
+    It would be helpful to check missing files before a heavy training run.
+
+    Args:
+        datalist: a list of data items, every item is a dictionary.
+            ususally generated by `load_decathlon_datalist` API.
+        keys: expected keys to check in the datalist.
+        root_dir: if not None, provides the root dir for the relative file paths in `datalist`.
+        allow_missing_keys: whether allow missing keys in the datalist items.
+            if False, raise exception if missing. default to False.
+
+    Returns:
+        A list of missing filenames.
+
+    """
+    missing_files = []
+    for item in datalist:
+        for k in ensure_tuple(keys):
+            if k not in item:
+                if not allow_missing_keys:
+                    raise ValueError(f"key `{k}` is missing in the datalist item: {item}")
+                continue
+
+            for f in ensure_tuple(item[k]):
+                if not isinstance(f, str):
+                    raise ValueError(f"filepath of key `{k}` must be a string or a list of strings, but got: {f}.")
+                if isinstance(root_dir, str):
+                    f = os.path.join(root_dir, f)
+                if not os.path.exists(f):
+                    missing_files.append(f)
```

Comment:
`os.path.exists` does not distinguish folder vs file, is that expected?

Reply:
Hi @wyli ,

Thanks for your review.
I think `os.path.exists` is OK for this feature, because `f` can also be a folder to load, for example, the DICOM series for 1 3D image.

Thanks.

Reply:
For example, if there's a folder named "test.img" but the datalist requires "test.img" file, the checking will be incorrect

Reply:
I see, I think that's very special case, it's not easy to know the expected item is a file or folder. Maybe we don't need to check that so far.

Thanks.